Makefile

```diff
@@ -2,23 +2,26 @@ all: build images
 
 # Define constants
 ##################
-BINDIR        ?= bin
-PLATFORM      ?= linux
-ARCH          ?= amd64
-CLUSTERAPI_BIN = $(BINDIR)/cluster-api
-VERSION       ?= $(shell git describe --always --abbrev=7 --dirty)
-GO_VERSION     = 1.10
-GO_BUILD       = env GOOS=$(PLATFORM) GOARCH=$(ARCH) go build -i $(GOFLAGS)
+BINDIR            ?= bin
+PLATFORM          ?= linux
+ARCH              ?= amd64
+CLUSTERAPI_BIN     = $(BINDIR)/cluster-api
+CLUSTERAPI_VERSION = $(shell git ls-remote https://github.com/kubernetes-sigs/cluster-api.git master | cut -c1-7)
+VERSION           ?= $(shell git describe --always --abbrev=7)
+GO_VERSION        ?= 1.10
+GO_BUILD           = env GOOS=$(PLATFORM) GOARCH=$(ARCH) go build -i $(GOFLAGS)
 
 AWS_MACHINE_CONTROLLER_PKG = github.com/enxebre/cluster-api-provider-aws
 
-DOCKER_CMD     = docker run --security-opt label:disable --rm -v $(PWD):/go/src/$(AWS_MACHINE_CONTROLLER_PKG) \
-                 -v $(PWD)/.pkg:/go/pkg buildimage
+DOCKER_CMD = docker run --security-opt label:disable --rm -v $(PWD):/go/src/$(AWS_MACHINE_CONTROLLER_PKG) -v $(PWD)/.pkg:/go/pkg buildimage
 
-
-CLUSTER_API_IMAGE = $(REGISTRY)cluster-api:$(VERSION)
-CONTROLLER_MANAGER_IMAGE = $(REGISTRY)controller-manager:$(VERSION)
-AWS_MACHINE_CONTROLLER_IMAGE = $(REGISTRY)aws-machine-controller:$(VERSION)
+MUTABLE_TAG                         ?= canary
```

Comment:
why do we need MUTABLE_TAG?

Reply:
We are currently tagging with first 7 characters of git SHA, MUTABLE_TAG lets us use `canary` as a tag for easier development. In the end it gives better developer experience